src/core/cast/closestPointToPointHybrid.template.js

```diff
@@ -0,0 +1,211 @@
+import { Vector3 } from 'three';
+import { COUNT, OFFSET, LEFT_NODE, RIGHT_NODE, IS_LEAF } from '../utils/nodeBufferUtils.js';
+import { BufferStack } from '../utils/BufferStack.js';
+import { ExtendedTrianglePool } from '../../utils/ExtendedTrianglePool.js';
+import { setTriangle } from '../../utils/TriangleUtilities.js';
+import { closestDistanceSquaredPointToBox } from '../utils/distanceUtils.js';
+import { SortedListDesc } from '../utils/SortedListDesc.js';
+
+const temp = /* @__PURE__ */ new Vector3();
+const temp1 = /* @__PURE__ */ new Vector3();
+const sortedList = new SortedListDesc();
+
+export function closestPointToPointHybrid/* @echo INDIRECT_STRING */(
+	bvh,
+	root,
+	point,
+	target,
+	sortedListMaxCount,
+	minThreshold,
+	maxThreshold
+) {
+
+	const minThresholdSq = minThreshold * minThreshold;
+	const maxThresholdSq = maxThreshold * maxThreshold;
+	let closestDistanceSq = Infinity;
+	let closestDistanceTriIndex = null;
+	BufferStack.setBuffer( bvh._roots[ root ] );
+
+	const { geometry } = bvh;
+	const { index } = geometry;
+	const pos = geometry.attributes.position;
+	const triangle = ExtendedTrianglePool.getPrimitive();
+
+	const { float32Array, uint16Array, uint32Array } = BufferStack;
+
+	if ( sortedListMaxCount === 0 ) {
+
+		_closestPointToPoint( 0 );
+
+	} else {
+
+		_closestPointToPointSorted();
+
+	}
+
+	BufferStack.clearBuffer();
+
+	if ( closestDistanceSq === Infinity ) return null;
+
+	const closestDistance = Math.sqrt( closestDistanceSq );
+
+	if ( ! target.point ) target.point = temp1.clone();
+	else target.point.copy( temp1 );
+	target.distance = closestDistance;
+	target.faceIndex = closestDistanceTriIndex;
+
+	return target;
+
+
+	function _closestPointToPointSorted() {
+
+		sortedList.clear();
+		let count = 0;
+		let node = { nodeIndex32: 0, distance: closestDistanceSquaredPointToBox( 0, float32Array, point ) };
+
+		do {
+
+			const { distance, nodeIndex32 } = node;
+
+			if ( distance >= closestDistanceSq ) return;
+
+			if ( count >= sortedListMaxCount ) {
+
+				if ( _closestPointToPoint( nodeIndex32 ) ) return;
+
+				continue;
+
+			}
+
+			count ++;
```

Comment:
Can you explain the logic and rationale of the hybrid function a bit? It looks like it will choose the "best" node from the first 16 nodes (by default) and then fall back to the basic approach of traversing to the leaves for each of those nodes in order. The count never goes down, right?